### PR TITLE
Raise an error during validation if a job has multiple gets for the same resource

### PR DESCRIPTION
**Pull request by cnelson**
Fixes https://github.com/concourse/concourse/issues/652

@vito This is the issue [we discussed earlier this week in slack](https://concourseci.slack.com/archives/general/p1478814623014485), feedback appreciated.

As a work of the United States Government, this PR is in the public domain within the United States. Additionally, we waive copyright and related rights in the work worldwide through the CC0 1.0 Universal public domain dedication.

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@cnelson Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/pivotal)!

[Click here](https://cla.pivotal.io/sync/pivotal?repositoryId=concourse/atc&pullRequestId=134) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/faq) for frequently asked questions.

File: config/validate.go [hunk 1]
```diff
@@ -218,9 +218,21 @@ func validateJobs(c atc.Config) ([]Warning, error) {
 			)
 		}
 
-		planWarnings, planErrMessages := validatePlan(c, identifier+".plan", atc.PlanConfig{Do: &job.Plan})
+		planInputs, planWarnings, planErrMessages := validatePlan(c, identifier+".plan", atc.PlanConfig{Do: &job.Plan})
```
Comment (vito) [inline config/validate.go]: Could this be done by looping over `JobInputs(job)` instead of collecting them during recursion?

**Comment by cnelson** [comment]
@vito updated this to use `JobInputs()`; thanks for the suggestion, it's much cleaner.

However us 18F folks still can't execute your CLA as it has no provision for work in the public domain which anything we do is.  Our folks reached out to `cla@pivotal.io` to discuss but we haven't heard back.

**Comment by cnelson** [comment]
Noting here that w heard back from the pivotal CLA team today and they said:

> Work is already underway to revise the CLA so that it accounts for the public domain use case you've raised.
In the meantime while this effort is underway, we can advise our project teams to accept your pull requests without CLA acceptance as long as you clearly mark your contributions with a public domain dedication.

**Comment by vito** [comment]
thanks for syncing up with our CLA folks! just a couple more fixups needed

**Comment by cnelson** [comment]
Updated!

**Comment by vito** [comment]
hmm could you rebase this? i'd like to see the build go green, and it looks like it's probably failing because it's a bit out of date.

**Comment by cnelson** [comment]
@vito This has been rebased, and the tests ran successfully.

**Comment by vito** [comment]
great, thanks!

**Pull request merged**